modules/exoplanets/tasks/exoplanetscsvtobintask.cpp

```diff
@@ -0,0 +1,677 @@
+        p.uTt = readFloatData(data);
+
+        getline(lineStream, data, ','); // TTREF
+        getline(lineStream, data, ','); // TTURL
+        getline(lineStream, data, ','); // V
+        getline(lineStream, data, ','); // VREF
+        getline(lineStream, data, ','); // VURL
+        getline(lineStream, data, ','); // VSINI
+        getline(lineStream, data, ','); // VSINIUPPER
+        getline(lineStream, data, ','); // VSINILOWER
+        getline(lineStream, data, ','); // UVSINI
+        getline(lineStream, data, ','); // VSINIREF
+        getline(lineStream, data, ','); // VSINIURL
+        getline(lineStream, data, ','); // KEPID
+        if (!data.empty()) {
+            isKeplerObject = true;
+        }
+        getline(lineStream, data); // KDE
+
+        if (!isKeplerObject) {
+            // calculate B-V from Teff if not exsisting
+            if (std::isnan(p.bmv)) {
+                p.bmv = bvFromTeff(teff);
+            }
+
+            // create look-up table
+            long pos = static_cast<long>(binFile.tellp());
+            planetName = speckStarname + " " + component;
+            lutFile << planetName << "," << pos << std::endl;
+            binFile.write(reinterpret_cast<char*>(&p), sizeof(Exoplanet));
+        }
+    }
+
+    progressCallback(1.f);
+}
+
+glm::vec3 ExoplanetsCsvToBinTask::starPosition(const std::string& starName) {
+    glm::vec3 position;
+    position[0] = NAN;
+    position[1] = NAN;
+    position[2] = NAN;
+    std::ifstream exoplanetsFile(_inputSpeckPath);
+    if (!exoplanetsFile) {
+        LERROR(fmt::format("Error opening file expl.speck."));
+    }
+
+    std::string line;
+    std::string data; // data
+    std::string name;
+    while (getline(exoplanetsFile, line)) {
+        bool shouldSkipLine = (
+            line.empty() || line[0] == '#' || line.substr(0, 7) == "datavar" ||
+            line.substr(0, 10) == "texturevar" || line.substr(0, 7) == "texture"
+            );
+
+        if (shouldSkipLine) {
+            continue;
+        }
+
+        std::istringstream linestream(line);
+        getline(linestream, data, '#');
+        getline(linestream, name);
+        name.erase(0, 1);
+
+        std::string coord;
+        if (name == starName) {
+            std::stringstream dataStream(data);
+            getline(dataStream, coord, ' ');
+            position[0] = std::stof(coord.c_str(), nullptr);
+            getline(dataStream, coord, ' ');
+            position[1] = std::stof(coord.c_str(), nullptr);
+            getline(dataStream, coord, ' ');
+            position[2] = std::stof(coord.c_str(), nullptr);
+            break;
+        }
+    }
+
+    // Apply transformation matrix to pos
+    glm::dmat4 _transformationMatrix = glm::dmat4(1.0);
+    glm::vec3 transformedPosition = glm::vec3(
+        _transformationMatrix * glm::dvec4(position, 1.0)
+    );
+
+    return transformedPosition;
+}
+
+float ExoplanetsCsvToBinTask::bvFromTeff(const float teff) {
```

Comment:
There is no real reason to declare an in-parameter as `const`  there is no optimization that can happen because of it and it makes too strong of a guarantee for when we wan to change the function declaration later;  let's say we change this function and all of a sudden we want to constrain `teff`.  So we either would have to make a copy that is `clamp`ed, or we would need to change the function definition toe `float teff` which might break client code unnecessarily